modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/actions/addFragment.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {fetch} from 'frontend-js-web';
+import {LAYOUT_DATA_ITEM_TYPES} from '../config/constants/layoutDataItemTypes';
+import {ADD_FRAGMENT_ENTRY_LINK_AND_ITEM} from './types';
+
+function addFragmentEntryLinkAndItem({
+	fragmentEntryLink,
+	itemId,
+	itemType,
+	parentId,
+	position
+}) {
+	const {
+		fragmentEntryLinkId
+	} = fragmentEntryLink;
+
+	return {
+		config: {
+			fragmentEntryLinkId,
+		},
+		fragmentEntryLinkId,
+		fragmentEntryLink,
+		itemId,
+		itemType,
+		parentId,
+		position,
+		type: ADD_FRAGMENT_ENTRY_LINK_AND_ITEM
+	};
+}
+
+export default function addFragment({
+	config,
+	store,
+	fragmentKey,
+	fragmentGroupId,
+	parentId,
+	position
+}) {
+	return (dispatch) => {
+		const {
+			portletNamespace
+		} = config;
+
+		const {
+			addFragmentEntryLinkURL,
+			classNameId,
+			classPK,
+			segmentsExperienceId
+		} = store;
+
+		const formData = new FormData();
+		formData.append(`${portletNamespace}fragmentKey`, fragmentKey);
+		formData.append(`${portletNamespace}classNameId`, classNameId);
+		formData.append(`${portletNamespace}classPK`, classPK);
+		formData.append(`${portletNamespace}groupId`, fragmentGroupId);
+		formData.append(`${portletNamespace}segmentsExperienceId`, segmentsExperienceId);
+
+		fetch(addFragmentEntryLinkURL, {
+			body: formData,
+			method: 'POST'
+		})
+			.then(response => response.json())
+			.then(fragmentEntryLink => {
+				// TODO: This is a temporary "hack"
+				//       until the backend is consitent
+				//       between both "metal+soy" and "react" versions
+				const {
+					content,
+				} = fragmentEntryLink;
+
+				fragmentEntryLink.content = {
+					value: {
+						content
+					}
+				};
+
+				dispatch(addFragmentEntryLinkAndItem({
+					fragmentEntryLink,
+					itemId: `thing-${Date.now()}`,
```

Comment:
This `itemId` is generated on the front end or it would be generated by the backend now that we are sending the information to the backend.

Reply:
For now it's generated on the frontend yes, I could have added a package for generating unique IDs (like `uuid` for example) but I thought it wasn't worth it just for a simple test.

As you'll have noticed in order to add an item, you first need to  

1 - Create a fragment entry link (generated on the backend)
2 - Add the item you want to create (update state and re-render)
3 - Ideally send the updated `layoutData` to the server  (backend - not done yet)

Concerning the backend/frontend communication I don't have all the details of the various workflows but maybe @p2kmgcl can help clarify all that.

Reply:
Thanks @julien!